protocols/delegate/contracts/Delegate.sol

```diff
@@ -169,17 +107,86 @@ contract Delegate is IDelegate, Ownable {
   function unsetRule(
     address _senderToken,
     address _signerToken
-  ) external onlyAdmin {
+  ) external onlyOwner {
+    unsetRuleInternal(
+      _senderToken,
+      _signerToken
+    );
+  }
 
-    // Delete the rule.
-    delete rules[_senderToken][_signerToken];
+  /**
+    * @notice sets a rule on the delegate and an intent on the indexer
+    * @dev only callable by owner
+    * @dev delegate needs to be given allowance from msg.sender for the _amountToStake
+    * @dev swap needs to be given permission to move funds from the delegate
+    * @param _senderToken the token the delgeate will send
+    * @param _senderToken the token the delegate will receive
+    * @param _rule the rule to set on a delegate
+    * @param _amountToStake the amount to stake for an intent
+    * @param _indexer the indexer the intent will be set on
+    */
+  function setRuleAndIntent(
+    address _senderToken,
+    address _signerToken,
+    Types.Rule calldata _rule,
+    uint256 _amountToStake,
+    IIndexer _indexer
+  ) external onlyOwner {
+    setRuleInternal(
+      _senderToken,
+      _signerToken,
+      _rule.maxSenderAmount,
+      _rule.priceCoef,
+      _rule.priceExp
+    );
 
-    emit UnsetRule(
+    require(
+      IERC20(_indexer.stakeToken())
+      .transferFrom(msg.sender, address(this), _amountToStake), "TRANSFER_FUNDS_ERROR"
+    );
+
+    //ensure that the indexer can pull funds from delegate account
+    require(
```

Comment:
We should be able to have this approval happen just one time in the constructor.